bpo-34206: Improve docs and test coverage for pre-init functions (#8023)
python/cpython: Misc/NEWS.d/next/Documentation/2018-07-04-20-35-25.bpo-34008.bqecIb.rst

> The ``Py_Main`` documentation moved from the "Very High Level API" section to the

Reviewer: ```suggestion
The :c:func:`Py_Main` documentation moved from the "Very High Level API" section to the
```

"Initialization and Finalization" section.

Also make it explicit that we expect ``Py_Main`` to typically be called instead
of ``Py_Initialize`` rather than after it (since ``Py_Main`` makes its own

> call to ``Py_Initialize``). However, also document that calling both is
> supported, but it's version dependent as to exactly which settings
> will be applied correctly in that case.

Reviewer: ```suggestion
call to ``Py_Initialize``). Document that calling both is
supported but is version dependent on which settings
will be applied correctly.
```
